cmd/depguard/main.go

```diff
@@ -79,23 +82,54 @@ func (*yamlConfigurator) parse(r io.Reader) (*depguard.LinterSettings, error) {
 }
 
 func getSettings() (*depguard.LinterSettings, error) {
-	fs := os.DirFS(".")
-	f, ft, err := findFile(fs)
+	fy, f, ft, err := findFile(".")
+	if errors.Is(err, fs.ErrNotExist) {
+		arg := []string{"list", "-f", "{{.Root -}}"}
+		out, cerr := exec.Command("go", arg...).Output() 
+		if cerr != nil {
+			return nil, cerr
+		}
+		fy, f, ft, err = findFile(strings.TrimRight(string(out), "\r\n"))
+	}
+	// careful: be sure to overwrite err (not shadow!) in the nested scope above ;)
 	if err != nil {
+		if e, ok := err.(*fs.PathError); ok {
+			err = e.Unwrap()
+		}
 		return nil, err
 	}
-	file, err := fs.Open(f)
+	file, err := fy.Open(f)
 	if err != nil {
 		return nil, fmt.Errorf("could not open %s to read: %w", f, err)
 	}
 	defer file.Close()
 	return ft.parse(file)
 }
 
-func findFile(fsys fs.FS) (string, configurator, error) {
+// The returned filepath is relative to given base path rel, or 
+// it is absolute if rel is empty or invalid.
+func caller(rel string) (name, f string, n int) {
+	if pc, _, _, ok := runtime.Caller(1); ok {
+		if fn := runtime.FuncForPC(pc); fn != nil {
+			name = fn.Name()
+			f, n = fn.FileLine(pc)
```

Comment:
This would get the source code path... not the current working directory of the executable. The example given only works because this is the main.go file and you assumed the current working directory to be in the same location as the main.go file.

Should be able to get using `os.Getwd()` and use the result to make `filepath.Rel` work.

Reply:
Yes, the source code path is what was intended. This method is used to generate error strings indicating in which function an error occurred. This isn't used for resolving paths related to the target project.

Reply:
This can probably be removed entirely anyway, as the error string it contributes to is never actually output. It was useful during debugging, but it serves no purpose now.

Reply:
Let's keep for debugging. Sorry for confusion, was trying to track variable names and all the different return values.